worker/src/request_init.rs

```diff
@@ -281,49 +297,134 @@ pub struct MinifyConfig {
 
 /// Configuration options for Cloudflare's image optimization feature:
 /// <https://blog.cloudflare.com/introducing-polish-automatic-image-optimizati/>
-#[wasm_bindgen]
-#[derive(Clone, Copy)]
+#[derive(Clone, Copy, Default, Serialize)]
+#[serde(rename_all = "kebab-case")]
 pub enum PolishConfig {
+    #[default]
     Off,
     Lossy,
     Lossless,
 }
 
-impl Default for PolishConfig {
-    fn default() -> Self {
-        Self::Off
-    }
+#[derive(Clone, serde::Serialize)]
+#[serde(untagged)]
+pub enum ResizeBorder {
+    Uniform {
+        color: String,
+        width: usize,
+    },
+    Varying {
+        color: String,
+        top: usize,
+        right: usize,
+        bottom: usize,
+        left: usize,
+    },
 }
 
-impl From<PolishConfig> for &str {
-    fn from(conf: PolishConfig) -> Self {
-        match conf {
-            PolishConfig::Off => "off",
-            PolishConfig::Lossy => "lossy",
-            PolishConfig::Lossless => "lossless",
-        }
-    }
+/// Configuration options for Cloudflare's image resizing feature:
+/// <https://developers.cloudflare.com/images/image-resizing/>
+#[derive(Clone, Default, serde::Serialize)]
+pub struct ResizeConfig {
```

Comment:
There's some more properties defined in the TS types that aren't here like `draw` and it'd be really nice if these fields were documented as it can be a little overwhelming seeing such a big struct.

https://workers-types.pages.dev/#RequestInitCfPropertiesImage should be handy for porting